pkg/api/types.go

```diff
@@ -1882,6 +1882,27 @@ const (
 
 	// DockerConfigJsonKey is the key of the required data for SecretTypeDockerConfigJson secrets
 	DockerConfigJsonKey = ".dockerconfigjson"
+
+	// SecretTypeBasicAuth contains data needed for basic authentication.
+	//
+	// Required at least one of fields:
+	// - Secret.Data["username"] - username used for authentication
+	// - Secret.Data["password"] - password or token needed for authentication
+	SecretTypeBasicAuth SecretType = "kubernetes.io/basic-auth"
+
+	// BasicAuthUsernameKey is the key of the username for SecretTypeBasicAuth secrets
+	BasicAuthUsernameKey = "username"
+	// BasicAuthPasswordKey is the key of the password or token for SecretTypeBasicAuth secrets
+	BasicAuthPasswordKey = "password"
+
+	// SecretTypeSSHAuth contains data needed for SSH authetication.
+	//
+	// Required field:
+	// - Secret.Data["ssh-privatekey"] - private SSH key needed for authentication
+	SecretTypeSSHAuth SecretType = "kubernetes.io/ssh-auth"
+
+	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
+	SSHAuthPrivateKey = "ssh-privatekey"
```

Comment:
If we have to pick an arbitrary value, and we have a pattern for key types,
it was primarily about picking something that clearly indicated what the
type was that fit into that (define a convention consistent with existing
conventions).

On Sat, Dec 19, 2015 at 7:22 AM, Jakub Hadvig <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/13853#discussion_r48091879>
> :
>
> > +	// - Secret.Data["password"] - password or token needed for authentication
> > +	SecretTypeBasicAuth SecretType = "kubernetes.io/basic-auth"
> > +
> > +	// BasicAuthUsernameKey is the key of the username for SecretTypeBasicAuth secrets
> > +	BasicAuthUsernameKey = "username"
> > +	// BasicAuthPasswordKey is the key of the password or token for SecretTypeBasicAuth secrets
> > +	BasicAuthPasswordKey = "password"
> > +
> > +	// SecretTypeSSHAuth contains data needed for SSH authetication.
> > +	//
> > +	// Required field:
> > +	// - Secret.Data["ssh-privatekey"] - private SSH key needed for authentication
> > +	SecretTypeSSHAuth SecretType = "kubernetes.io/ssh-auth"
> > +
> > +	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
> > +	SSHAuthPrivateKey = "ssh-privatekey"
>
> As @smarterclayton <https://github.com/smarterclayton> already explained:
> For ssh-privatekey, "privatekey" is a common identifier used by SSH
> config, that's the only reason.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13853/files#r48091879>.
>